src/services/recim/participant.ts

```diff
@@ -6,10 +6,22 @@ export type Participant = {
   Username: string;
   Email: string;
   Role: string;
+  AllowEmails: boolean;
+  SpecifiedGender: string;
   GamesAttended: number;
   Status: string;
   Notification: ParticipantNotification[];
   IsAdmin: boolean;
+  IsCustom: boolean;
+  FirstName: string;
+  LastName: string;
+};
+
+type CustomParticipant = {
+  AllowEmails: boolean;
+  SpecifiedGender: string;
+  FirstName: string;
+  LastName: string;
 };
```

Comment:
Take a look at what we do for `activity`.ts, we can cut down on some of these repeated pieces of data. Also for the record @Kevin-Chu58's API PR MAKES a username that ensures no duplicates. However, I can see a small chance that there could be overlap if a Gordon User is added i.e:

silas.white (exists)
we add another silas.white -> silas.white2
Gordon adds another silas.white -> silas.white2 (conflict)

Reply:
The autogenerated username algorithm is now fixed. We will always add a suffix `.custom` after each non-Gordon participant username.